runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1909,11 +1909,10 @@ MM_CopyForwardScheme::copy(MM_EnvironmentVLHGC *env, MM_AllocationContextTarok *
 		UDATA hotFieldPadSize = 0;
 		MM_CopyScanCacheVLHGC *copyCache = NULL;
 		void *newCacheAlloc = NULL;
-		bool doesObjectNeedHash = false;
 		GC_ObjectModel *objectModel = &_extensions->objectModel;
 		
 		/* Object is in the evacuate space but not forwarded. */
-		_extensions->objectModel.calculateObjectDetailsForCopy(env, forwardedHeader, &objectCopySizeInBytes, &objectReserveSizeInBytes, &doesObjectNeedHash);
+		_extensions->objectModel.calculateObjectDetailsForCopy(env, forwardedHeader, &objectCopySizeInBytes, &objectReserveSizeInBytes);
```

Comment:
there is local objectModel created, just above.

Reply:
And latter we do seem to fetch instanceHotFieldDescription. Seems like we could have only one variant of calculateObjectDetailsForCopy()

For some reason, how we use hotField is a bit different between Scavenge and CopyForward - we don't check if it's an array of a particular size limit!?

Reply:
I don't understand why there is that difference either. Is it time to start pushing the other PRs that remove this hot field work? This would simplify code in various ways and simply allow for only one variant of calculateObjectDetailsForCopy, in addition to the bigger benefit of removing a performance slowing "optimization".

Reply:
The fact that CopyForward and Scavenge differ how they use instanceHotFieldDescription does not seem to prevent us that we have only one variant of calculateObjectDetailsForCopy(), the one that returns instanceHotFieldDescription, right?

Reply:
Yes you are correct, this would not prevent us from having only one variant now.

Reply:
ok, let's remove one. Have CopyForward gets instanceHotFieldDescription from calculateObjectDetailsForCopy, like scavenger